src/Pollers/DeviceMappingPoller.php

```diff
@@ -86,13 +86,22 @@ public function poll(array $work):array
                 foreach ($myChunksWithDeviceType as $hostWithDeviceType)
                 {
                     try {
+						$time_start = microtime(true);
                         $device = new Device();
                         $device->setId($hostWithDeviceType['id']);
                         $device->setSnmpObject($this->buildSnmpObjectForHost($hostWithDeviceType));
 
                         //Additional 'case' statements can be added here to break out querying to a separate device mapper
                         $mapper = $this->getDeviceMapper($hostWithDeviceType, $device);
-                        $device = $mapper->mapDevice();
+                        $device = $mapper->mapDevice;
+						
+						//figure out if we're running overtime to poll devices
+						$time_end = microtime(true); 
+						$device->setTimer($time_end-$time_start);
+						
+						if($device->getTimer() > 5){
+							$this->log->log("{$hostWithDeviceType['ip']} took longer than 5 seconds to poll",Logger::ERROR);
+						}
```

Comment:
Can you wrap this in a debug mode check?

Reply:
added debug checks